sale_partner_version/tests/test_sale_partner_version.py

```diff
@@ -10,10 +10,64 @@ class TestSalePartnerVersion(TransactionCase):
     def setUp(self):
         super(TestSalePartnerVersion, self).setUp()
         self.sale = self.env.ref('sale.sale_order_1')
+        partner_vals = {
+            'name': u'Name',
+            'street': u'Street',
+            'street2': u'Street2',
+            'zip': u'Zip',
+            'city': u'City',
+            'country_id': self.env.ref('base.fr').id}
+        self.partner = self.env['res.partner'].create(partner_vals)
 
     def test_sale_version_partner(self):
-        self.assertFalse(self.sale.partner_invoice_id.version_hash)
-        self.assertFalse(self.sale.partner_shipping_id.version_hash)
+        partner = self.sale.partner_invoice_id
+        partner.write({'city': 'new city'})
+        self.sale.invalidate_cache()
+        self.assertEqual(self.sale.partner_invoice_id, partner)
+        self.assertNotEqual(self.sale.partner_invoice_id.parent_id, partner)
+        self.assertEqual(self.sale.partner_shipping_id, partner)
+        self.assertNotEqual(self.sale.partner_shipping_id.parent_id, partner)
         self.sale.action_confirm()
-        self.assertTrue(self.sale.partner_invoice_id.version_hash)
-        self.assertTrue(self.sale.partner_shipping_id.version_hash)
+        partner.write({'city': 'city update'})
+        self.assertNotEqual(self.sale.partner_invoice_id, partner)
```

Comment:
We should test that the 

self.assertEqual(self.sale.partner_id, partner)


The contact should be not archived